pkg/cluster/deploybaseresources_additional.go

```diff
@@ -92,6 +92,20 @@ func (m *manager) clusterServicePrincipalRBAC() *arm.Resource {
 	)
 }
 
+func (m *manager) fpspStorageBlobContributorRBAC(storageAccountName, principalID string) (*arm.Resource, error) {
+	if !m.doc.OpenShiftCluster.UsesWorkloadIdentity() {
+		return nil, fmt.Errorf("fpspStorageBlobContributorRBAC called for a Cluster Service Principal cluster")
+	}
+	resourceTypeStorageAccount := "Microsoft.Storage/storageAccounts"
+	return rbac.ResourceRoleAssignmentWithName(
+		rbac.RoleStorageBlobDataContributor,
+		"'"+principalID+"'",
```

Comment:
minor: Using 2 different string manipulations in the same area, suggestion to keep consistency 
```suggestion
		fmt.Sprintf("'%s'", principalID)
```

Reply:
Done.